xcm-support/src/lib.rs

```diff
@@ -75,3 +78,86 @@ impl UnknownAsset for () {
 		Err(DispatchError::Other(NO_UNKNOWN_ASSET_IMPL))
 	}
 }
+
+/// Extracts the `AccountId32` from the passed `location` if the network
+/// matches.
+pub struct RelaychainAccountId32Aliases<Network, AccountId>(PhantomData<(Network, AccountId)>);
+impl<Network: Get<NetworkId>, AccountId: From<[u8; 32]> + Into<[u8; 32]> + Clone>
+	xcm_executor::traits::Convert<MultiLocation, AccountId> for RelaychainAccountId32Aliases<Network, AccountId>
+{
+	fn convert(location: MultiLocation) -> Result<AccountId, MultiLocation> {
+		let id = match location {
+			MultiLocation {
+				parents: 1,
+				interior: X1(AccountId32 {
+					id,
+					network: NetworkId::Any,
+				}),
+			} => id,
+			MultiLocation {
+				parents: 1,
+				interior: X1(AccountId32 { id, network }),
+			} if network == Network::get() => id,
+			_ => return Err(location),
+		};
+		Ok(id.into())
+	}
+
+	fn reverse(who: AccountId) -> Result<MultiLocation, AccountId> {
+		Ok((
+			1,
+			AccountId32 {
+				id: who.into(),
+				network: Network::get(),
+			},
+		)
+			.into())
+	}
+}
+
+/// when Relay an XCM message from a given `interior` location, if the given
+/// `interior` is not `Here`, the destination will receive a xcm message
+/// beginning with `DescendOrigin` as the first instruction. so the xcm message
+/// format must match this order:
+/// `DescendOrigin`,`WithdrawAsset`,`BuyExecution`,`Transact`.
+pub struct AllowEquivalentAccountsFrom<T>(PhantomData<T>);
```

Comment:
The same is still a bit confusing. `AllowEquivalentAccountsFrom<Everything> ` is not actually allow from everything.

Maybe `AllowEquivalentParentAccountsFrom`

Reply:
ok. I was thinking add restricted condition to DescendOrigin, only allow `DescendOrigin(X1(Junction::AccountId32 { network} if network == Network.get()` and Network trait bound is `RelayNetwork`

Reply:
I also change `Everything` to a new struct `IsParent` to only allow `Relaychain` is the `origin`.

Reply:
I think we may not need `T: Contains<MultiLocation>` trait bound, we can use more strict condition inside like 
```rust
ensure!(origin.contains_parents_only(1));
```